homeassistant/components/ecovacs/entity.py

```diff
@@ -120,3 +121,35 @@ class EcovacsCapabilityEntityDescription(
     """Ecovacs entity description."""
 
     capability_fn: Callable[[Capabilities], CapabilityEntity | None]
+
+
+class EcovacsLegacyEntity(Entity):
+    """Ecovacs legacy bot entity."""
+
+    _attr_has_entity_name = True
+    _attr_should_poll = False
+
+    def __init__(self, device: VacBot) -> None:
+        """Initialize the legacy Ecovacs entity."""
+        self.device = device
+        vacuum = device.vacuum
+
+        self.error: str | None = None
+        self._attr_unique_id = vacuum["did"]
+
+        if (name := vacuum.get("deviceName")) is None:
```

Comment:
while comparing the data fields with the from the representation in the app, i figured out, that `nick` isn't set via the app, but `deviceName` ist the shown name of the bot in the app (_it just defaults to the product family and model name_), so i've changed the used data fields for name and model in the device info